Tweaks how some tongues have no taste

coiax opened the pull request:
Skeletons, abductors and ethereals have no sense of taste. Previously
they would taste something "indescribable", but instead, they will not
taste anything or get any message. This also means they will no longer
get mood buffs from eating/drinking high quality food.

:cl: coiax
tweak: Skeletons, abductors, etherals and anyone without a tongue
can no longer taste anything, or get food/drink moodlets.
tweak: People with the ageusia quirk will also not get any taste
messages, or get food/drink moodlets.
/:cl:

- The utility item "taster" has had some additional messages added.
- Adds a name and description to snail tongues, and colours them
  brown in leu of a proper sprite.

coiax committed:
Tweaks how some tongues have no taste

Skeletons, abductors and ethereals have no sense of taste. Previously
they would taste something "indescribable", but instead, they will not
taste anything or get any message. This also means they will no longer
get mood buffs from eating/drinking high quality food.

Carbons without tongues also can no longer taste anything.

- The utility item "taster" has had some additional messages added.

coiax committed:
Remove can_taste proc

Instead of a single proc that is only used, so carbons can override it
with the missing tongue, just have carbons unable to taste anything by
default, and then have the tongue "supress" that.

RaveRadbury commented:
Is this being applied to ageusia as well?

coiax commented:
Yes, people with ageusia will also not taste anything.

LemonInTheDark commented on code/modules/mob/living/taste.dm:
```diff
@@ -4,21 +4,31 @@
+
+	var/taste_sensitivity = get_taste_sensitivity()
+
 	if(last_taste_time + 50 < world.time)
-		var/taste_sensitivity = get_taste_sensitivity()
-		var/text_output = from.generate_taste_message(taste_sensitivity,src)
+		var/text_output = from.generate_taste_message(src, taste_sensitivity)
```
Why call this before the cooldown?

coiax replied:
Do you mean why call `get_taste_sensitivity()` before the cooldown check?

LemonInTheDark replied:
yes

LemonInTheDark commented on code/modules/reagents/chemistry/holder.dm:
```diff
@@ -924,8 +924,9 @@
  *
  * Arguments:
  * * minimum_percent - the lower the minimum percent, the more sensitive the message is.
+ * * mob/living/taster - who is doing the tasting. Some mobs can pick up specific flavours.
```
You should make the order of the args match the comment, or vis versua

LemonInTheDark commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -8,6 +8,11 @@
 	attack_verb_simple = list("lick", "slobber", "slap", "french", "tongue")
 	var/list/languages_possible
 	var/say_mod = null
+	/*
+	* Whether the owner of this tongue can taste anything. Being
+	* set to FALSE will mean no taste feedback will be provided.
+	*/
```
This isn't the proper format for autodoc comments

coiax replied:
Is there a method of doing DMDOC multiline variable comments, or do I just use a series of `///` comments?

LemonInTheDark replied:
I'm honestly not sure, I believe a series of `///` will break, one liners are best, but if more then one is needed your best bet is the proc comment format

LemonInTheDark commented on code/modules/mob/living/taste.dm:
```diff
@@ -4,21 +4,31 @@
 	var/last_taste_time
 	var/last_taste_text
 
+/*
+* Gets the "taste_sensitivity" of a given mob. This is used in calculating
+* what flavours the mob can pick up, with a lower number closer to 0
+* being better.
+*/
```
Nor is this I don't think

LemonInTheDark reviewed (changes requested):
Do you have maint approval for the species changes?

coiax committed:
Modify with code review feedback

- A function is only called when its value will be used.
- Order of arguments in a docstring now matches the reality
- A docstring has been tweaked to be in the DMDOC style.

coiax committed:
Merge branch 'master' into capable-of-tasting-flavour

coiax committed:
Rework variable docstring

LemonInTheDark commented on code/modules/mob/living/taste.dm:
```diff
@@ -4,21 +4,32 @@
 	var/last_taste_time
 	var/last_taste_text
 
+/*
+ * Gets taste sensitivity of given mob
+ *
+ * This is used in calculating what flavours the mob can pick up,
+ * with a lower number being able to pick up more distinct flavours.
+ */
```
```suggestion
/**
 * Gets taste sensitivity of given mob
 *
 * This is used in calculating what flavours the mob can pick up,
 * with a lower number being able to pick up more distinct flavours.
 */
```

Learner10 commented:
As an ethereal main, this bothers me, especially as mine was often a detective who'd enjoy a donut and a drink, if at least for the sizzle and texture. I was never able to explain the flavor of things, only that I just liked them, which I was ok with.

Commit:
Update code/modules/mob/living/taste.dm

Co-authored-by: LemonInTheDark <58055496+LemonInTheDark@users.noreply.github.com>
